### PR TITLE
DGS-10129: Update RunRequirements that overwrite SR cluster command annotation

### PR DESCRIPTION
**Pull request by GunalKupta**
Release Notes
-------------

Breaking Changes
- PLACEHOLDER

New Features
- PLACEHOLDER

Bug Fixes
- no need to run login a second time to run SR cluster commands

Checklist
---------
- [N/A] Leave this box unchecked if features are not yet available in production

What
----

As identified in an Always-On Governance [Bug bash](https://confluentinc.atlassian.net/wiki/spaces/DG/pages/3324280858/Always-On+Bug+Bash#Bugs), after running the following commands with v4 CLI:
```
confluent login
confluent environment create aog_cli_bug
confluent environment use <new-id>
confluent kafka cluster create cluster_0 --cloud gcp --region us-central1
confluent sr cluster describe
```
... the sr cluster describe command returns the following error:
```
Error: failed to validate Schema Registry client: 401 Unauthorized
```
Updating the annotation on the SR cluster describe command (to the same as parent) fixed this behavior. The parent command (`sr cluster`) annotation matches the annotation on the `environment` and `kafka cluster` commands.

References
----------
[DGS-10129](https://confluentinc.atlassian.net/browse/DGS-10129)

Test & Review
-------------

[DGS-10129]: https://confluentinc.atlassian.net/browse/DGS-10129?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by akhileshm1 — COMMENTED** [review]
shouldn't this change go to `main` as well?

**Comment by GunalKupta** [comment]
> shouldn't this change go to `main` as well?

Sure, but something to keep in consideration: The issue was not reported in v3 during the bug bash. I also noticed that the  SR cluster enable, delete, and upgrade commands also use the old annotation (`pcmd.RequireCloudLogin`) so they would all need to be updated to match.

File: internal/schema-registry/command_cluster_describe.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func (c *command) newClusterDescribeCommand() *cobra.Command {
 		Short:       "Describe the Schema Registry cluster for this environment.",
 		Args:        cobra.NoArgs,
 		RunE:        c.clusterDescribe,
-		Annotations: map[string]string{pcmd.RunRequirement: pcmd.RequireCloudLogin},
+		Annotations: map[string]string{pcmd.RunRequirement: pcmd.RequireNonAPIKeyCloudLogin},
```
Comment (sgagniere) [inline internal/schema-registry/command_cluster_describe.go:47]: We should also the annotation of `internal/schema-registry/command_cluster.go` to `RequireNonAPIKeyCloudLoginOrOnPremLogin` to match.

**Review by sgagniere — COMMENTED** [review]
This change is fine because we should be using the NonAPIKey annotations here.

But this shouldn't be causing the error from the bug bash. The annotation only affects whether or not you're able to run the command. That specific error is returned by `GetSchemaRegistryClient`, which is called inside of `clusterDescribe`. That means the annotation didn't block the command execution.

If possible, could we get `--unsafe-trace` logs for this bug?

**Comment by GunalKupta** [comment]
Command: `dist/confluent_$(go env GOOS)_$(go env GOARCH)/confluent sr cluster describe --unsafe-trace`

```
GET /srcm/v2/clusters?environment=env-stgcp5jz85 HTTP/1.1
Host: api.stag.cpdev.cloud
User-Agent: Confluent-CLI/v3.48.1-SNAPSHOT-9a484286 (https://confluent.io; support@confluent.io)
Accept: application/json
Authorization: Bearer <Redacted>
Accept-Encoding: gzip


2024-02-21T17:52:19.573-0600 [DEBUG] performing request%!(EXTRA string=method, string=GET, string=url, *url.URL=https://api.stag.cpdev.cloud/srcm/v2/clusters?environment=env-stgcp5jz85)
2024/02/21 17:52:20
HTTP/2.0 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Headers: Authorization,Accept,Origin,DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Content-Range,Range,X-Client-Request-Id,X-Correlation-Id
Access-Control-Allow-Methods: GET,POST,OPTIONS,PUT,DELETE,PATCH
Content-Type: application/json
Date: Wed, 21 Feb 2024 23:52:20 GMT
Deprecation: Mon, 04 Mar 2024 00:00:00 UTC
Server: nginx
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Envoy-Upstream-Service-Time: 119
X-Frame-Options: deny
X-Ratelimit-Limit: 20
X-Ratelimit-Remaining: 19
X-Ratelimit-Reset: 1
X-Xss-Protection: 1; mode=block

{
  "api_version": "srcm/v2",
  "data": [
    {
      "api_version": "srcm/v2",
      "id": "lsrc-stgc8d8yk5",
      "kind": "Cluster",
      "metadata": {
        "created_at": "2024-02-21T23:51:59.366391Z",
        "resource_name": "crn://confluent.cloud/organization=73485678-d50c-4979-9b6b-1de13f7476af/environment=env-stgcp5jz85/schema-registry=lsrc-stgc8d8yk5",
        "self": "https://api.stag.cpdev.cloud/srcm/v2/clusters/lsrc-stgc8d8yk5",
        "updated_at": "2024-02-21T23:51:59.366391Z"
      },
      "spec": {
        "display_name": "Always On Stream Governance Package",
        "environment": {
          "id": "env-stgcp5jz85",
          "related": "https://api.stag.cpdev.cloud/org/v2/environments/env-stgcp5jz85",
          "resource_name": "crn://confluent.cloud/organization=73485678-d50c-4979-9b6b-1de13f7476af/environment=env-stgcp5jz85"
        },
        "http_endpoint": "https://psrc-3p76j.us-central1.gcp.stag.cpdev.cloud",
        "package": "ESSENTIALS",
        "region": {
          "id": "sgreg-3",
          "related": "https://api.stag.cpdev.cloud/srcm/v2/regions/sgreg-3",
          "resource_name": "crn://confluent.cloud/schema-registry-region=sgreg-3"
        }
      },
      "status": {
        "phase": "PROVISIONED"
      }
    }
  ],
  "kind": "ClusterList",
  "metadata": {
    "first": "https://api.stag.cpdev.cloud/srcm/v2/clusters",
    "total_size": 1
  }
}

2024/02/21 17:52:20
GET /srcm/v2/regions/sgreg-3 HTTP/1.1
Host: api.stag.cpdev.cloud
User-Agent: Confluent-CLI/v3.48.1-SNAPSHOT-9a484286 (https://confluent.io; support@confluent.io)
Accept: application/json
Authorization: Bearer <Redacted>
Accept-Encoding: gzip


2024-02-21T17:52:20.116-0600 [DEBUG] performing request%!(EXTRA string=method, string=GET, string=url, *url.URL=https://api.stag.cpdev.cloud/srcm/v2/regions/sgreg-3)
2024/02/21 17:52:20
HTTP/2.0 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Headers: Authorization,Accept,Origin,DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Content-Range,Range,X-Client-Request-Id,X-Correlation-Id
Access-Control-Allow-Methods: GET,POST,OPTIONS,PUT,DELETE,PATCH
Content-Type: application/json
Date: Wed, 21 Feb 2024 23:52:20 GMT
Deprecation: Mon, 04 Mar 2024 00:00:00 UTC
Server: nginx
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Envoy-Upstream-Service-Time: 31
X-Frame-Options: deny
X-Ratelimit-Limit: 20
X-Ratelimit-Remaining: 18
X-Ratelimit-Reset: 1
X-Xss-Protection: 1; mode=block

{
  "api_version": "srcm/v2",
  "id": "sgreg-3",
  "kind": "Region",
  "metadata": {
    "created_at": "2022-08-24T00:57:19.455023Z",
    "resource_name": "crn://confluent.cloud/schema-registry-region=sgreg-3",
    "self": "https://api.stag.cpdev.cloud/srcm/v2/regions/sgreg-3",
    "updated_at": "2022-08-24T00:57:19.455023Z"
  },
  "spec": {
    "cloud": "GCP",
    "display_name": "Iowa (us-central1)",
    "packages": [
      "ESSENTIALS",
      "ADVANCED"
    ],
    "region_name": "us-central1"
  }
}

2024/02/21 17:52:20
GET /srcm/v2/clusters?environment=env-stgcp5jz85 HTTP/1.1
Host: api.stag.cpdev.cloud
User-Agent: Confluent-CLI/v3.48.1-SNAPSHOT-9a484286 (https://confluent.io; support@confluent.io)
Accept: application/json
Authorization: Bearer <Redacted>
Accept-Encoding: gzip


2024-02-21T17:52:20.290-0600 [DEBUG] performing request%!(EXTRA string=method, string=GET, string=url, *url.URL=https://api.stag.cpdev.cloud/srcm/v2/clusters?environment=env-stgcp5jz85)
2024/02/21 17:52:20
HTTP/2.0 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Headers: Authorization,Accept,Origin,DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Content-Range,Range,X-Client-Request-Id,X-Correlation-Id
Access-Control-Allow-Methods: GET,POST,OPTIONS,PUT,DELETE,PATCH
Content-Type: application/json
Date: Wed, 21 Feb 2024 23:52:20 GMT
Deprecation: Mon, 04 Mar 2024 00:00:00 UTC
Server: nginx
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Envoy-Upstream-Service-Time: 53
X-Frame-Options: deny
X-Ratelimit-Limit: 20
X-Ratelimit-Remaining: 17
X-Ratelimit-Reset: 1
X-Xss-Protection: 1; mode=block

{
  "api_version": "srcm/v2",
  "data": [
    {
      "api_version": "srcm/v2",
      "id": "lsrc-stgc8d8yk5",
      "kind": "Cluster",
      "metadata": {
        "created_at": "2024-02-21T23:51:59.366391Z",
        "resource_name": "crn://confluent.cloud/organization=73485678-d50c-4979-9b6b-1de13f7476af/environment=env-stgcp5jz85/schema-registry=lsrc-stgc8d8yk5",
        "self": "https://api.stag.cpdev.cloud/srcm/v2/clusters/lsrc-stgc8d8yk5",
        "updated_at": "2024-02-21T23:51:59.366391Z"
      },
      "spec": {
        "display_name": "Always On Stream Governance Package",
        "environment": {
          "id": "env-stgcp5jz85",
          "related": "https://api.stag.cpdev.cloud/org/v2/environments/env-stgcp5jz85",
          "resource_name": "crn://confluent.cloud/organization=73485678-d50c-4979-9b6b-1de13f7476af/environment=env-stgcp5jz85"
        },
        "http_endpoint": "https://psrc-3p76j.us-central1.gcp.stag.cpdev.cloud",
        "package": "ESSENTIALS",
        "region": {
          "id": "sgreg-3",
          "related": "https://api.stag.cpdev.cloud/srcm/v2/regions/sgreg-3",
          "resource_name": "crn://confluent.cloud/schema-registry-region=sgreg-3"
        }
      },
      "status": {
        "phase": "PROVISIONED"
      }
    }
  ],
  "kind": "ClusterList",
  "metadata": {
    "first": "https://api.stag.cpdev.cloud/srcm/v2/clusters",
    "total_size": 1
  }
}

2024/02/21 17:52:21
GET / HTTP/1.1
Host: psrc-3p76j.us-central1.gcp.stag.cpdev.cloud
User-Agent: Confluent-CLI/v3.48.1-SNAPSHOT-9a484286 (https://confluent.io; support@confluent.io)
Accept: application/vnd.schemaregistry.v1+json,application/vnd.schemaregistry+json; qs=0.9,application/json; qs=0.5
Authorization: Bearer <Redacted>
Target-Sr-Cluster: lsrc-stgc8d8yk5
Accept-Encoding: gzip


2024-02-21T17:52:21.036-0600 [DEBUG] performing request%!(EXTRA string=method, string=GET, string=url, *url.URL=https://psrc-3p76j.us-central1.gcp.stag.cpdev.cloud/)
2024/02/21 17:52:21
HTTP/2.0 401 Unauthorized
Connection: close
Content-Type: text/plain
Date: Wed, 21 Feb 2024 23:52:21 GMT
X-Request-Id: a642a579-63d1-4333-afe5-037f3726f1ab

Bearer realm="SchemaRegistry",error="invalid_token"
Error: failed to validate Schema Registry client: 401 Unauthorized
```

**Comment by GunalKupta** [comment]
Closing this PR in favor of merging changes to main branch instead of v4 for now: https://github.com/confluentinc/cli/pull/2642

**Pull request closed without merging**